### PR TITLE
Add header component

### PR DESCRIPTION
**Pull request by Karolk99**

### CODE REVIEW

File: assets/js/utils/headerUtils.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    start_date: "",
+    presenters: [],
+  };
+};
+
+export const getEventInfo = (setEventInfo: React.Dispatch<React.SetStateAction<EventInfo>>) => {
+  axios
+    .get("/resources/webinars/" + getChannelId())
+    .then((response) => {
+      console.log(response.data);
```
Comment (Rados13) [inline assets/js/utils/headerUtils.ts:20]: Do we need this?

File: assets/js/utils/headerUtils.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      const start_date = response.data.webinar.start_date.replace("T", " ");
+      setEventInfo({ ...response.data.webinar, start_date: start_date });
+    })
+    .catch(() => {
+      alert("Couldn't get event information. Please reload this page.");
+    });
```
Comment (Rados13) [inline assets/js/utils/headerUtils.ts:26]: Maybe we should log this error?

File: assets/js/pages/Event.tsx [hunk 3]
```diff
@@ -72,16 +36,12 @@ const Event = () => {
...
-  }, []);
-
   useEffect(() => {
     const alreadyJoined = eventChannel?.state === "joined";
     if (name && !alreadyJoined) {
-      const channel = socket.channel(`event:${eventInfo.link}`, {
+      const channel = socket.channel(`event:${getChannelId()}`, {
         name: name,
-        isModerator: eventInfo.isModerator,
+        isModerator: true,
```
Comment (Rados13) [inline assets/js/pages/Event.tsx:44]: Are you sure that this should be always true?

File: assets/js/pages/Event.tsx [hunk 4]
```diff
@@ -108,11 +70,7 @@ const Event = () => {
           </div>
         </div>
         <div className="Participants">
-          <ParticipantsList
-            username={name}
-            isModerator={eventInfo.isModerator}
-            eventChannel={eventChannel}
-          />
+          <ParticipantsList username={name} isModerator={true} eventChannel={eventChannel} />
```
Comment (Rados13) [inline assets/js/pages/Event.tsx:73]: Same as above.

File: assets/js/utils/channelUtils.ts [hunk 5]
```diff
@@ -102,3 +103,11 @@ export const syncPresenters = (
...
   }
 };
+
+export const getChannelId = (): string => window.location.pathname.split("/")[2];
+
+const compareParticipants = (x: Participant, y: Participant): number => {
+  if (x.isModerator != y.isModerator) return x.isModerator ? -1 : 1;
+  if (x.isPresenter != y.isPresenter) return x.isPresenter ? -1 : 1;
+  return 0;
+};
```
Comment (Rados13) [inline assets/js/utils/channelUtils.ts:113]: I think it is too complex I would create a function e.g: `participantToNumber` which takes the participant and return rank of this participant. Then this function could look like this:
```suggestion
const compareParticipants = (x: Participant, y: Participant): number => {
return participantToNumber(x) - participantToNumber(y)
};
```

File: assets/js/utils/headerUtils.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  if (eventChannel) {
+    const presence = new Presence(eventChannel);
+
+    const updateParticipantsNumber = () => {
+      let counter = 0;
+      presence.list((name: string, metas: any) => {
+        counter += 1;
+      });
+      setParticipantsNumber(counter);
+    };
```
Comment (Rados13) [inline assets/js/utils/headerUtils.ts:39]: I think this way it is simpler.
```suggestion
    const updateParticipantsNumber = () => {
      setParticipantsNumber(presence.list().length);
    };
```

**Review by Rados13 — CHANGES_REQUESTED** [review]

File: assets/js/pages/Event.tsx (hunk 3 above)
Comment (Karolk99) [reply assets/js/pages/Event.tsx:44]: We don't have eventInfo state in Event component, isModerator will be added in the next pull request. Right now value is mocked.

File: assets/js/pages/Event.tsx (hunk 4 above)
Comment (Karolk99) [reply assets/js/pages/Event.tsx:73]: same answer

File: assets/js/utils/headerUtils.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+};
+
+export const getEventInfo = (setEventInfo: React.Dispatch<React.SetStateAction<EventInfo>>) => {
+  axios
+    .get("/resources/webinars/" + getChannelId())
+    .then((response) => {
+      const start_date = response.data.webinar.start_date.replace("T", " ");
+      setEventInfo({ ...response.data.webinar, start_date: start_date });
+    })
+    .catch(() => {
```
Comment (Rados13) [inline assets/js/utils/headerUtils.ts:23]: Log error which we receive in catch.

**Review by Rados13 — APPROVED** [review]

**Pull request merged**